### PR TITLE
Core & Internals: cleanup old heartbeats. Closes #4459

### LINKED ISSUE #4459: Heartbeat cleanup
Motivation
----------
Due to the more dynamic `hostnames` with kubernetes the current cleanup functions of heartbeats are not sufficient, and the table is polluted with old heartbeats which never get removed. Heartbeats do continue to work normally, but the table will grow indefinitely.

Modification
------------
Add a cleanup of all heartbeats older than 24h to the `sanity_check` method.

### PR DESCRIPTION
**Pull request by rcarpa**
Due to the more dynamic hostnames with kubernetes the current cleanup
functions of heartbeats are not sufficient, and the table is polluted
with old heartbeats which never get removed.

Add a cleanup of all heartbeats older than 24h.

### CODE REVIEW

File: lib/rucio/core/heartbeat.py [hunk 1]
```diff
@@ -70,6 +83,9 @@ def _sanity_check(executable, hostname, hash_executable=None, session=None):
             if not pid_exists(pid):
                 session.query(Heartbeats).filter_by(hostname=hostname, pid=pid).delete()
 
+    if expiration_delay:
+        cardiac_arrest(older_than=expiration_delay, session=session)
```
Comment (bari12) [inline lib/rucio/core/heartbeat.py:87]: I wonder if we should put this into a try/except. Reason being that the `sanity_check` is often called in a daemon outside of the while loop which would recover from errors. And there might be some row-contention/deadlock errors if this is called at the same time.

Comment (rcarpa) [reply lib/rucio/core/heartbeat.py:87]: _sanity_check is wrapped by sanity_check() which does exactly that. Do we need to catch any exceptions other than "databaseError" ?

Comment (bari12) [reply lib/rucio/core/heartbeat.py:87]: Right, I have misread this, ignore what I said :-)

**Review by mlassnig — APPROVED** [review]

**Comment by bari12** [comment]
Will be included in `1.25` release line

**Pull request merged**